### PR TITLE
Editor styles: cache transform

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Editor styles settings remain constant for all previews, so we should cache the transforms so the work is not repeated for each preview.

## Why?

1st run: -5% (load patterns)
2nd run -5.6%

### CODE REVIEW

File: packages/block-editor/src/utils/transform-styles/index.js [hunk 1]
```diff
@@ -5,6 +5,52 @@ import postcss, { CssSyntaxError } from 'postcss';
 import wrap from 'postcss-prefixwrap';
 import rebaseUrl from 'postcss-urlrebase';
 
+const transformStylesCache = new WeakMap();
```
Comment (youknowriad) [inline packages/block-editor/src/utils/transform-styles/index.js:8]: WeakMap is such a magic tool :P

File: packages/block-editor/src/utils/transform-styles/index.js [hunk 2]
```diff
@@ -18,45 +64,14 @@ import rebaseUrl from 'postcss-urlrebase';
  * @return {Array} converted rules.
  */
 const transformStyles = ( styles, wrapperSelector = '' ) => {
```
Comment (youknowriad) [inline packages/block-editor/src/utils/transform-styles/index.js:66]: As mentioned in person, if we're using an iframe, transforming styles might not be necessary anymore assuming we also pass the "url" to the stylesheet in addition to its content. We could just use the URL to load the stylesheets within the iframe.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

File: packages/block-editor/src/utils/transform-styles/index.js (hunk 2 above)
Comment (ellatrix) [reply packages/block-editor/src/utils/transform-styles/index.js:66]: Right, I'll try this next. This PR is still useful for an editor without iframe though (when there's a registered block that is not v3)

File: packages/block-editor/src/utils/transform-styles/index.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-editor/src/utils/transform-styles/index.js:8]: Yes, very!

**Comment by t-hamano** [comment]
If the post editor is not iframed, I think the CSS selector will be `.editor-styles-wrapper` instead of `body`. If we cache the style, won't this selector also be loaded in iframe editor instances such as preview?

This seems to cause problems in certain scenarios. Please see [this comment](https://github.com/WordPress/gutenberg/issues/60990#issuecomment-2094806820).

**Pull request merged**